tool/src/dart/class.rs

```diff
@@ -363,23 +364,62 @@ impl<'a, 'dartcx, 'tcx: 'dartcx> TyGenContext<'a, 'dartcx, 'tcx> {
 
         let return_ty = self.gen_dart_return_type_name(&method.output);
         let method_name = self.cx.formatter.fmt_method_name(method);
+
         let declaration = if method.param_self.is_none() {
             if return_ty == type_name {
-                if method_name == "new" {
+                let mut method_name = &*method_name;
+                for prefix in ["create", "new", "default"] {
+                    method_name = method_name.strip_prefix(prefix).unwrap_or(method_name);
+                }
+                if method_name.is_empty() {
                     format!("factory {type_name}({params})")
                 } else {
-                    format!("factory {type_name}.{method_name}({params})")
+                    format!(
+                        "factory {type_name}.{}({params})",
+                        method_name.to_lower_camel_case()
+                    )
                 }
-            } else if method_name == "new" {
-                format!("static {return_ty} new_({params})")
+            } else if params.is_empty()
+                && (return_ty != "bool" || method_name.starts_with("is"))
+                && !matches!(method.output, hir::ReturnType::Fallible(..))
+            {
+                format!(
+                    "static late final {return_ty} {method_name} = \
+                        _capi<ffi.NativeFunction<{ffi_return_ty} Function()>>('{c_method_name}')\
+                        .asFunction<{ffi_cast_return_ty} Function()>(isLeaf: true)();"
+                )
             } else {
                 format!("static {return_ty} {method_name}({params})")
             }
-        // } else if method.params.is_empty() && return_ty != "void" && method_name != "toString" {
-        //     format!("{return_ty} get {method_name}")
-        // } else if method_name.starts_with("set") && method.params.len() == 1 {
-        //     use heck::ToLowerCamelCase;
-        //     format!("{return_ty} set {}({params})", method_name.strip_prefix("set").unwrap().to_lower_camel_case())
+        } else if method.params.is_empty()
```

Comment:
nit: I think this logic belongs in the `fmt()` family of functions

Reply:
I disagree, this is pretty fundamental to what the method is, it's not just a cosmetic change.